### PR TITLE
178 list units in base management

### LINKED ISSUE #178: List units in barracks
**Describe the solution you'd like**
List units as entries
- Name
- Role
- Squad they belong to
- City they are currently deployed to

blick #179

### PR DESCRIPTION
**Pull request by cthomas68**
Fixes #178 .

Changes proposed in this pull request:
- list units in barracks view
- improve GUI

bug note: scroll bar doesn't work

### CODE REVIEW

**Review by Toeecko — DISMISSED** [review]
Looks amazing!

File: Assets/Scripts/Strategy/BaseManagement/Buildings/Barracks.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        [SerializeField] private GameObject unitListPanel;
+        [SerializeField] private DisplayModelController displayModelController;
+        [SerializeField] private DisplayDataController displayDataController;
+
+        private IList<IUnit> activeUnitsList;
+
+        public GameObject TrainUnitCanvas
+        {
+            get { return trainUnitCanvas; }
+        }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Buildings/Barracks.cs]: Use the `=>` operator here

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public class GenericBuilding : MonoBehaviour, IBuilding
+    {
+        [SerializeField] private GameObject mainCanvas;
+
+        protected GenericStrongholdCell Cell;
+
+        public GameObject MainCanvas
+        {
+            get { return mainCanvas; }
+        }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs]: Use the `=>` operator here

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using SwordAndBored.Strategy.BaseManagement.Buildings;
+using UnityEngine;
+
+namespace SwordAndBored.StrategyView.BaseManagement.Buildings
+{
+    public class GenericBuilding : MonoBehaviour, IBuilding
+    {
+        [SerializeField] private GameObject mainCanvas;
+
+        protected GenericStrongholdCell Cell;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericStrongholdCell.cs [hunk 4]
```diff
@@ -2,19 +2,12 @@
...
 
-        public GameObject Canvas
-        {
-            get { return canvas; }
-            set { value = canvas; }
-        }
-
         public Button ClickableObject
         {
             get { return clickableObject; }
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Buildings/GenericStrongholdCell.cs:13]: Use the `=>` operator here

File: Assets/Scripts/Strategy/BaseManagement/CameraManager.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿using UnityEngine;
+
+namespace SwordAndBored.Strategy.BaseManagement
+{
+    public class CameraManager : MonoBehaviour
+    {
+        [SerializeField] private Transform StagingArea;
+        [SerializeField] private Transform DefaultPosition;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/CameraManager.cs]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/ConfirmUnitListener.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+namespace SwordAndBored.Strategy.BaseManagement.Units
+{
+    public class ConfirmUnitListener : MonoBehaviour
+    {
+        public UnitFactory UnitFactory;
+        public InputField UnitName;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/ConfirmUnitListener.cs:9]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/DisplayDataController.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        [SerializeField] private TMP_Text CurrentHP;
+        [SerializeField] private TMP_Text Initiative;
+        [SerializeField] private TMP_Text PhysicalAttack;
+        [SerializeField] private TMP_Text PhysicalDefense;
+        [SerializeField] private TMP_Text MagicAttack;
+        [SerializeField] private TMP_Text MagicDefense;
+        [SerializeField] private TMP_Text Role;
+        [SerializeField] private TMP_Text Name;
+        [SerializeField] private TMP_Text Squad;
+        [SerializeField] private TMP_Text Town;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/DisplayDataController.cs:19]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/RoleEntryDisplay.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace SwordAndBored.Strategy.BaseManagement.Units
+{
+    public class RoleEntryDisplay : MonoBehaviour
+    {
+        public RoleEntry RoleEntry;
+        public Text EntryRole;
+        public Text EntryDescription;
+        public Image EntryImage;
+
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/RoleEntryDisplay.cs]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntry.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+using UnityEngine;
+
+namespace SwordAndBored.Strategy.BaseManagement.Units
+{
+    public class UnitEntry : ScriptableObject
+    {
+        public string UnitName;
+        public string Role;
+        public string CurrentTown;
+        public string CurrentSquad;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/UnitEntry.cs]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    public class UnitEntryDisplay : MonoBehaviour
+    {
+        public UnitEntry UnitEntry;
+
+        public Image UnitImage;
+        public TMP_Text UnitName;
+        public TMP_Text Squad;
+        public TMP_Text Town;
+
+        public List<Sprite> RoleSpriteList;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs]: Member variables should be camel case

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+        public void SetDisplay()
+        {
+            IDictionary<string, Sprite> RoleImageDictionary = new Dictionary<string, Sprite>()
+        {
+            {"Warrior", RoleSpriteList[0]},
+            {"Scout", RoleSpriteList[1]},
+            {"Mage", RoleSpriteList[2]}
+        };
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs]: Bad indentation

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitFactory.cs [hunk 12]
```diff
@@ -1,18 +1,17 @@
...
-namespace SwordAndBored.Strategy.BaseManagement
+namespace SwordAndBored.Strategy.BaseManagement.Units
 {
     public class UnitFactory : MonoBehaviour
     {
         public string UnitRole { get; set; }
         public string UnitName { get; set; }
 
         public GameObject NameUnitCanvas;
+        public Barracks Barracks;
```
Comment (wynn-rj) [inline Assets/Scripts/Strategy/BaseManagement/Units/UnitFactory.cs]: Member variable should be camel case

**Review by wynn-rj — CHANGES_REQUESTED** [review]
For all of your objects it would probably be good to use the `AssertHelper.IsSetInEditor()` to identify which objects should be set in the editor but it is not needed for me to approve

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitFactory.cs (hunk 12 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Units/UnitFactory.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs (hunk 11 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs (hunk 10 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Units/UnitEntryDisplay.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Units/UnitEntry.cs (hunk 9 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Units/UnitEntry.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Units/RoleEntryDisplay.cs (hunk 8 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Units/RoleEntryDisplay.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/CameraManager.cs (hunk 5 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/CameraManager.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericStrongholdCell.cs (hunk 4 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Buildings/GenericStrongholdCell.cs:13]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs (hunk 3 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs (hunk 2 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Buildings/GenericBuilding.cs]: 4725445

File: Assets/Scripts/Strategy/BaseManagement/Buildings/Barracks.cs (hunk 1 above)
Comment (cthomas68) [reply Assets/Scripts/Strategy/BaseManagement/Buildings/Barracks.cs]: 4725445

**Review by wynn-rj — APPROVED** [review]

**Review by Toeecko — APPROVED** [review]

**Pull request merged**